apps/api/src/app/integrations/usecases/deactivate-integration/deactivate-integration.usecase.ts

```diff
@@ -1,11 +1,24 @@
 import { Injectable } from '@nestjs/common';
 import { IntegrationRepository } from '@novu/dal';
+import { FeatureFlagCommand, GetFeatureFlag } from '../../../feature-flags/use-cases';
 import { DeactivateSimilarChannelIntegrationsCommand } from './deactivate-integration.command';
 
 @Injectable()
 export class DeactivateSimilarChannelIntegrations {
-  constructor(private integrationRepository: IntegrationRepository) {}
+  constructor(private integrationRepository: IntegrationRepository, private getFeatureFlag: GetFeatureFlag) {}
   async execute(command: DeactivateSimilarChannelIntegrationsCommand): Promise<void> {
+    const shouldKeepIntegrationsActive = await this.getFeatureFlag.isMultiProviderConfigurationEnabled(
+      FeatureFlagCommand.create({
+        environmentId: command.environmentId,
+        organizationId: command.organizationId,
+        userId: command.userId,
+      })
+    );
+
+    if (shouldKeepIntegrationsActive) {
+      return;
+    }
+
```

Comment:
any tests?

Reply:
Yes, test is coming. This is still WIP as you see above ;) Forgot to change to draft pr and can not do it after creation :(